Unable to build: error: invalid use of incomplete type ‘class QObject’

I have been running into this problem:
```
In file included from ../databasedatatypes.cpp:19:0:
../databasedatatypes.h:27:22: error: invalid use of incomplete type ‘class QObject’
 class Enums : public QObject
                      ^
In file included from /usr/include/qt/QtCore/qmetatype.h:45:0,
                 from /usr/include/qt/QtCore/QMetaType:1,
                 from /usr/include/qt/QtPositioning/qgeocoordinate.h:37,
                 from /usr/include/qt/QtPositioning/QGeoCoordinate:1,
                 from ../databasedatatypes.h:25,
                 from ../databasedatatypes.cpp:19:
/usr/include/qt/QtCore/qobjectdefs.h:247:7: note: forward declaration of ‘class QObject’
 class QObject;
       ^
../databasedatatypes.cpp: In constructor ‘Enums::Enums(QObject*)’:
../databasedatatypes.cpp:23:5: error: type ‘QObject’ is not a direct base of ‘Enums’
     QObject(parent)
     ^
Makefile:2294: recipe for target 'databasedatatypes.o' failed
make: *** [databasedatatypes.o] Error 1
make: *** Waiting for unfinished jobs....
```

I think all of the dependencies are installed, the qt5 package group is installed:
https://www.archlinux.org/groups/x86_64/qt5/
The build command is:
```
mkdir build && cd build && qmake-qt5 -o Makefile .. && make -j 5
```

Version 1.0.0-stable

Same problem when I try to compile git version on Gentoo 64 bits with same commands:

```
...
g++ -c -pipe -O2 -Wall -W -D_REENTRANT -fPIC -DNATIVE_ASEMAN_NOTIFICATION -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_SQL_LIB -DQT_POSITIONING_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_CORE_LIB -I../../Papyrus -I. -I../qtdropbox -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtQuick -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtSql -isystem /usr/include/qt5/QtPositioning -isystem /usr/include/qt5/QtQml -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtCore -I. -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o searchhighlighter.o ../searchhighlighter.cpp
g++ -c -pipe -O2 -Wall -W -D_REENTRANT -fPIC -DNATIVE_ASEMAN_NOTIFICATION -DQT_NO_DEBUG -DQT_QUICK_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_SQL_LIB -DQT_POSITIONING_LIB -DQT_QML_LIB -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_CORE_LIB -I../../Papyrus -I. -I../qtdropbox -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtQuick -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtSql -isystem /usr/include/qt5/QtPositioning -isystem /usr/include/qt5/QtQml -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtCore -I. -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o databasedatatypes.o ../databasedatatypes.cpp
In file included from ../databasedatatypes.cpp:19:0:
../databasedatatypes.h:27:22: error: uso inválido del tipo incompleto ‘class QObject’
 class Enums : public QObject
                      ^
In file included from /usr/include/qt5/QtCore/qmetatype.h:45:0,
                 from /usr/include/qt5/QtCore/QMetaType:1,
                 from /usr/include/qt5/QtPositioning/qgeocoordinate.h:37,
                 from /usr/include/qt5/QtPositioning/QGeoCoordinate:1,
                 from ../databasedatatypes.h:25,
                 from ../databasedatatypes.cpp:19:
/usr/include/qt5/QtCore/qobjectdefs.h:247:7: error: declaración adelantada de ‘class QObject’
 class QObject;
       ^
../databasedatatypes.cpp: En el constructor ‘Enums::Enums(QObject*)’:
../databasedatatypes.cpp:23:5: error: el tipo ‘QObject’ no es una base directa de ‘Enums’
     QObject(parent)
     ^
Makefile:2486: fallo en las instrucciones para el objetivo 'databasedatatypes.o'
make: *** [databasedatatypes.o] Error 1
```

```
QMake version 3.0
Using Qt version 5.5.1 in /usr/lib64
```

--- a/databasedatatypes.h
+++ b/databasedatatypes.h
@@ -23,6 +23,7 @@
 #include <QColor>
 #include <QStringList>
 #include <QGeoCoordinate>
+#include <QObject>
 
 class Enums : public QObject
 {

